torch/csrc/jit/ir.cpp

```diff
@@ -1138,6 +1138,31 @@ void Node::removeAllInputs() {
   inputs_.clear();
 }
 
+void Node::permuteInputs(const std::vector<size_t>& new_order) {
+  schema_ = nullptr;
+  AT_ASSERT(new_order.size() == inputs_.size());
+  std::vector<Value*> new_inputs;
+  new_inputs.reserve(new_order.size());
+  for (size_t i = 0; i < new_order.size(); ++i) {
+    new_inputs.push_back(inputs_[new_order[i]]);
+    auto it = findUseForInput(new_order[i]);
+    it->offset = i;
+  }
+  inputs_ = std::move(new_inputs);
```

Comment:
In both loops, can you assert that a value is not being used twice. This can be done by blanking out the Value* in the original list, and checking that the value is not-null on lookup.

Reply:
I can check that an index is not being used twice, but since this is being used on node inputs a value can appear multiple times